crates/cargo-test-support/src/registry.rs

```diff
@@ -864,6 +866,20 @@ impl Package {
         self
     }
 
+    /// Call with `true` to publish a git dependency in a "local registry".
+    ///
+    /// The difference between this and [Package::local] is that this will
+    /// skip checksum generation as git dependencies do not have checksums.
+    ///
+    /// See `source-replacement.html#local-registry-sources` for more details
+    /// on local registries. See `local_registry.rs` for the tests that use
+    /// this.
+    pub fn local_from_git(&mut self, local: bool) -> &mut Package {
```

Comment:
This names the function for a specific policy.  I think I'd prefer if we focused on the mechanism being provided and just made a `skip_checksum` function`.